pkg/depsolvednf/depsolvednf_test.go

```diff
@@ -335,7 +334,9 @@ func TestSolverDepsolveAll(t *testing.T) {
 			assert := assert.New(t)
 			tc := testCases[tcName]
 
-			res, err := solver.DepsolveAll(tc.packageSets, tc.sbomType)
+			solver := NewSolver("platform:el9", "9", "x86_64", "rhel9.0", tmpdir)
+			solver.SBOMType(tc.sbomType)
```

Comment:
I would love us to move away from this pattern of constructing an instance just to modify it in the next step. Ideally moving towards the functional options pattern.

Reply:
Yeah the functional options you showed me the other day looks really nice.